test/e2e/virtual/apiexport/authorizer_test.go

```diff
@@ -98,131 +95,198 @@ func TestAPIExportAuthorizers(t *testing.T) {
+						IdentityHash:  sherriffsIdentityHash,
+						All:           true,
+					},
 				},
 			},
 		},
-	}
-	_, err = user2KcpClient.Cluster(serviceProvider2Path).ApisV1alpha1().APIExports().Create(ctx, cowboysAPIExport, metav1.CreateOptions{})
-	require.NoError(t, err)
 
-	t.Logf("grant user-3 to be able to bind cowboys API export \"today-cowboys\" from workspace %q", serviceProvider2Path)
-	cr, crb = createClusterRoleAndBindings(
-		"user-3-bind",
-		"user-3", "User",
-		[]string{"bind"},
-		"apis.kcp.io", "apiexports", "today-cowboys",
-	)
-	_, err = kubeClient.Cluster(serviceProvider2Path).RbacV1().ClusterRoles().Create(ctx, cr, metav1.CreateOptions{})
-	require.NoError(t, err)
-	_, err = kubeClient.Cluster(serviceProvider2Path).RbacV1().ClusterRoleBindings().Create(ctx, crb, metav1.CreateOptions{})
-	require.NoError(t, err)
+		&rbacv1.ClusterRole{
+			ObjectMeta: metav1.ObjectMeta{Name: "user-3-bind"},
+			Rules: []rbacv1.PolicyRule{
+				{APIGroups: []string{"apis.kcp.io"}, ResourceNames: []string{"today-cowboys"}, Resources: []string{"apiexports"}, Verbs: []string{"bind"}},
+			},
+		},
+		&rbacv1.ClusterRoleBinding{
+			ObjectMeta: metav1.ObjectMeta{Name: "user-3-bind"},
+			Subjects:   []rbacv1.Subject{{Kind: "User", Name: "user-3"}},
+			RoleRef:    rbacv1.RoleRef{APIGroup: rbacv1.SchemeGroupVersion.Group, Kind: "ClusterRole", Name: "user-3-bind"},
+		},
 
-	t.Logf("Create an APIBinding in consumer workspace %q that points to the today-cowboys export from %q", tenantPath, serviceProvider2Path)
-	apiBinding := &apisv1alpha1.APIBinding{
-		ObjectMeta: metav1.ObjectMeta{
-			Name: "cowboys",
+		&rbacv1.ClusterRole{
+			ObjectMeta: metav1.ObjectMeta{Name: "user-3-maximum-permission-policy"},
+			Rules: []rbacv1.PolicyRule{
+				{APIGroups: []string{"wildwest.dev"}, Resources: []string{"cowboys"}, Verbs: []string{"delete", "create", "list", "watch", "get", "patch"}},
+			},
 		},
-		Spec: apisv1alpha1.APIBindingSpec{
-			Reference: apisv1alpha1.BindingReference{
-				Export: &apisv1alpha1.ExportBindingReference{
-					Path: serviceProvider2Path.String(),
-					Name: "today-cowboys",
+		&rbacv1.ClusterRoleBinding{
+			ObjectMeta: metav1.ObjectMeta{Name: "user-3-maximum-permission-policy"},
+			Subjects:   []rbacv1.Subject{{Kind: "User", Name: "apis.kcp.io:binding:user-3"}},
+			RoleRef:    rbacv1.RoleRef{APIGroup: rbacv1.SchemeGroupVersion.Group, Kind: "ClusterRole", Name: "user-3-maximum-permission-policy"},
+		},
+	))
+
+	t.Logf("bind cowboys and claimed sherriffs in the tenant workspace %q", tenantPath)
+	require.NoError(t, apply(t, ctx, tenantPath, framework.UserConfig("user-3", rest.CopyConfig(cfg)),
+		&apisv1alpha1.APIBinding{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "wild.wild.west",
+			},
+			Spec: apisv1alpha1.APIBindingSpec{
+				Reference: apisv1alpha1.BindingReference{
+					Export: &apisv1alpha1.ExportBindingReference{
+						Path: serviceProvider1Path.String(),
```

Comment:
nit: if `serviceProvider1` is for sherrifs and `serviceProvider2` is cowboys, can we update to `sherrifsProvider` and `cowboysProvider`?

Reply:
oh sure